pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStage.java

```diff
@@ -102,31 +116,37 @@ public String toGroovy() {
         if (environment != null) {
             result.append(environment.toGroovy());
         }
-        result.append("steps {\n");
-        if (branches.size() > 1) {
-            result.append("parallel(");
-            boolean first = true;
-            for (ModelASTBranch branch: branches) {
-                if (first) {
-                    first = false;
-                } else {
-                    result.append(',');
+        if (branches.isEmpty() && parallelStages != null) {
+            result.append("parallelStages {\n");
```

Comment:
This does not follow the RFC.
And revenge of the camelCase :) Why not go with `parallel` as the RFC states? Confusion? There are plenty of synonyms or other contextually applickable things to `parallel` to choose from if you don't want to use that word; `threads`, `simultaneous`, `concurrent`. But I personally think that reusing the word `stages` is best. It self explains in a way that it is sub stages to this stage although that could add to the meaning as it is not apparent they would run in parallel so it would change the implementation a bit since probably an attribute should then be added to declare them to be executed in parallel.